python/k8s.py

```diff
@@ -29,14 +29,32 @@ def __init__(self):
         self.context = None
 
     def call(self, arguments):
-        call(['kubectl', '--context', self.context] + arguments)
+        context = self.current_context()
+        if context != None:
+            call(['kubectl', '--context', context] + arguments)
+        else:
+            call(['kubectl'] + arguments)
 
     def output(self, arguments):
-        return check_output(['kubectl', '--context', self.context] + arguments)
+        context = self.current_context()
+        if context != None:
+            return check_output(['kubectl', '--context', context] + arguments)
+        else:
+            return check_output(['kubectl'] + arguments)
+
+    def current_context(self):
```

Comment:
in python to add some logic in a property, we use a @property annotation and rename the actual property value with a leading underscore:

~~~python
@property
def context(self):
     # do some stuff
    retun self._context
~~~

Reply:
Tks, fixed